### PR TITLE
FLEX-4364 ~ Adds IEC61850 documentation for switch devices.

### PR DESCRIPTION
**Pull request by kevinsmeets**

### CODE REVIEW

File: Protocols/IEC61850/SWDevice-010805/StartSelfTest.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+## StartSelfTest messages
+
+### Description
+
+Request which commands the device to switch all light relays on and then queries the device for the status of the relays.
+Part of this request is also to retrieve the status of the relays after switching on.
```
Comment (rlemmers) [inline Protocols/IEC61850/SWDevice-010805/StartSelfTest.md]: More or less duplicates "and then queries the device for the status of the relays", which is on the previous line.

File: Protocols/IEC61850/SWDevice-010805/StopSelfTest.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
+## StopSelfTest messages
+
+### Description
+
+Request which commands the device to switch all light relays off and then queries the device for the status of the relays.
+Part of this request is also to retrieve the status of the relays after switching off.
```
Comment (rlemmers) [inline Protocols/IEC61850/SWDevice-010805/StopSelfTest.md]: More or less duplicates "and then queries the device for the status of the relays", which is on the previous line.

File: Protocols/IEC61850/SWDevice-010805/SetEventNotifications.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+Request which commands a device to set its EventNotification mask.
+
+Response which returns the result of the request.
+
+### Message definitions
+
+|**ATTRIBUTE**|**FC**|**SUB ATTRIBUTE**|**DATATYPE**|**DESCRIPTION**|
+|---|---|---|---|---|
+|CSLC.EvnBuf|CF|enbEvnType|VisString32|Bitmask indicating which event notification types are enabled, all event types enabled: "11FFFFFF".|
```
Comment (rlemmers) [inline Protocols/IEC61850/SWDevice-010805/SetEventNotifications.md]: Is 11FFFFFF correct?

File: Protocols/IEC61850/SWDevice-010805/UpdateDeviceSslCertification.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+### Description
+
+Request to download a new SSL certificate from the certificate server. The device will be given the domain name and URL where the certificate is located.
+
+### IEC61850 Fields
+
+|**ATTRIBUTE**|**FC**|**SUB ATTRIBUTE**|**DATATYPE**|**DESCRIPTION**|
+|---|---|---|---|---|
+|CSLC.CARepl|CF|url|VisString255|Set new CA file download URL here, device will download the new CA file and then replace the old CA file at startT.|
+|CSLC.CARepl|CF|startT|VisString32|Device will switch CA file at this Timestamp.|
```
Comment (rlemmers) [inline Protocols/IEC61850/SWDevice-010805/UpdateDeviceSslCertification.md]: Time and date? Or only time?

File: Protocols/IEC61850/SWDevice-010805/UpdateFirmware.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+Request which commands a device to download and install new firmware. The request contains a URL defining the location of the new firmware image. The device should download the firmware from that location.
+
+Response which returns the result of the request.
+
+### IEC61850 Fields
+
+|**ATTRIBUTE**|**FC**|**SUB ATTRIBUTE**|**DATATYPE**|**DESCRIPTION**|
+|---|---|---|---|---|
+|CSLC.FuncFwDw|CF|url|VisString255|Functional firmware. Set new firmware file download URL here, device will download the new firmware file and then replace the old firmwrae file at startT.|
```
Comment (rlemmers) [inline Protocols/IEC61850/SWDevice-010805/UpdateFirmware.md]: firmwrae => firmware

File: Protocols/IEC61850/SWDevice-010805/UpdateFirmware.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+|CSLC.FuncFwDw|CF|startT|Timestamp|Functional firmware. Device will try to install new firmware file at this Timestamp.|
+|CSLC.ScyFwDw|CF|url|VisString255|Security firmware. Set new firmware file download URL here, device will download the new firmware file and then replace the old firmwrae file at startT.|
```
Comment (rlemmers) [inline Protocols/IEC61850/SWDevice-010805/UpdateFirmware.md]: firmwrae => firmware

File: Protocols/IEC61850/FlexOVL_540_171101_2/EventNotification.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
+## EventNotification messages
+
+### Description
+
+Light sensor device can send buffered report containing the current state of the digital inputs. The buffered report will be sent on data change.
+OSGP will interpret the buffered report and save event information contained in the reports.
```
Comment (rlemmers) [inline Protocols/IEC61850/FlexOVL_540_171101_2/EventNotification.md]: Provide some information about the content of an event, which is created in OSGP (data to the platform triggered by the event).

File: Protocols/IEC61850/SWDevice-010805/EventNotification.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
+## EventNotification messages
+
+### Description
+
+Buffered report sent from device to platform containing information about 1 event.
```
Comment (rlemmers) [inline Protocols/IEC61850/SWDevice-010805/EventNotification.md]: Provide some information about the content of an event, which is created in OSGP (data to the platform triggered by the event).

File: Protocols/IEC61850/SWDevice-010805/SetTransition.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
+## SetTransition messages
+
+### Description
+
+Request which commands a device to switch all relays configured as light relays according to  a [light schedule-entry](SetSchedule.md).
```
[the remark this answers is not in the document]
Comment (kevinsmeets) [reply Protocols/IEC61850/SWDevice-010805/SetTransition.md]: Was "(SetSchedule.md)" meant as a reference? If so, change it to [SetSchedule.md]. -> it is a link to the SetSchedule documentation page. The format is correct `(click me)[./folder/file.md]`.

File: Protocols/IEC61850/SWDevice-010805/SetEventNotifications.md (hunk 3 above)
Comment (kevinsmeets) [reply Protocols/IEC61850/SWDevice-010805/SetEventNotifications.md]: No, it should be `1FFFFFF`.

File: Protocols/IEC61850/FlexOVL_540_171101_2.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
+## Contract
```
[the remark this answers is not in the document]
Comment (kevinsmeets) [reply Protocols/IEC61850/FlexOVL_540_171101_2.md]: Done.

File: Protocols/IEC61850/SWDevice-010805.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
+## Contract
```
[the remark this answers is not in the document]
Comment (kevinsmeets) [reply Protocols/IEC61850/SWDevice-010805.md]: Done.

File: Protocols/IEC61850/FlexOVL_540_171101_2/GetStatus.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+Soap requests and responses sent to and from platform:
+``` xml
+<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" 
+xmlns:ns="http://www.opensmartgridplatform.org/schemas/common/2014/10"
+xmlns:ns1="http://www.opensmartgridplatform.org/schemas/publiclighting/adhocmanagement/2014/10">
+   <soapenv:Header>
+      <ns:ApplicationName>SoapUI</ns:ApplicationName>
+      <ns:UserName>Kevin</ns:UserName>
+      <ns:OrganisationIdentification>LianderNetManagement</ns:OrganisationIdentification>
```
[the remark this answers is not in the document]
Comment (kevinsmeets) [reply Protocols/IEC61850/FlexOVL_540_171101_2/GetStatus.md]: We are using the same organization identification in the OSLP documentation. Not changing it for now.

File: Protocols/IEC61850/SWDevice-010805/GetStatus.md [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+
+Response which returns the result of the request and, if 'result = OK', contains the current status for all of the relays and other information.
+
+### IEC61850 Fields
+
+|**ATTRIBUTE**|**FC**|**SUB ATTRIBUTE**|**DATATYPE**|**DESCRIPTION**|
+|---|---|---|---|---|
+|XSWC1.Pos|ST|stVal|BOOLEAN|Current switch status for relay 1.|
+|XSWC2.Pos|ST|stVal|BOOLEAN|Current switch status for relay 2.|
+|XSWC3.Pos|ST|stVal|BOOLEAN|Current switch status for relay 3.|
```
[the remark this answers is not in the document]
Comment (kevinsmeets) [reply Protocols/IEC61850/SWDevice-010805/GetStatus.md]: Done.

File: Protocols/IEC61850/SWDevice-010805/SetConfiguration.md [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+
+Response which returns the result of the request.
+
+### IEC61850 Fields
+
+|**ATTRIBUTE**|**FC**|**SUB ATTRIBUTE**|**DATATYPE**|**DESCRIPTION**|
+|---|---|---|---|---|
+|XSWC1.SwType|CO|Oper.ctlVal|INT8|Switch type for relay 1, tariff = 0, light = 1.|
+|XSWC2.SwType|CO|Oper.ctlVal|INT8|Switch type for relay 2, tariff = 0, light = 1.|
+|XSWC3.SwType|CO|Oper.ctlVal|INT8|Switch type for relay 3, tariff = 0, light = 1.|
```
[the remark this answers is not in the document]
Comment (kevinsmeets) [reply Protocols/IEC61850/SWDevice-010805/SetConfiguration.md]: Done.

File: Protocols/IEC61850/SWDevice-010805/SetLight.md [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+In case the value 'index' is included, multiple instances of LightValue can be used (up to 6), each indicating a particular relay.
+
+Response which returns the result of the request.
+
+### IEC61850 Fields
+
+|**ATTRIBUTE**|**FC**|**SUB ATTRIBUTE**|**DATATYPE**|**DESCRIPTION**|
+|---|---|---|---|---|
+|XSWC2.Pos|CO|Oper.ctlVal|BOOLEAN|Flag which, if set to true, immediately switches relay 2 on.|
+|XSWC3.Pos|CO|Oper.ctlVal|BOOLEAN|Flag which, if set to true, immediately switches relay 3 on.|
```
[the remark this answers is not in the document]
Comment (kevinsmeets) [reply Protocols/IEC61850/SWDevice-010805/SetLight.md]: Done.

File: Protocols/IEC61850/SWDevice-010805/SetSchedule.md [hunk 16]
```diff
@@ -0,0 +1,561 @@
...
+
+Request which commands a device to set a light or tariff schedule.
+
+Response which returns the result of the request.
+
+### IEC61850 Fields
+
+|**ATTRIBUTE**|**FC**|**SUB ATTRIBUTE**|**DATATYPE**|**DESCRIPTION**|
+|---|---|---|---|---|
+|XSWC1.Sche|CF|sche1.enable|BOOLEAN|Flag indicating the schedule entry is enabled.|
```
[the remark this answers is not in the document]
Comment (kevinsmeets) [reply Protocols/IEC61850/SWDevice-010805/SetSchedule.md]: Done.

File: Protocols/IEC61850/SWDevice-010805/EventNotification.md (hunk 8 above)
Comment (kevinsmeets) [reply Protocols/IEC61850/SWDevice-010805/EventNotification.md]: Done.

**Pull request merged**